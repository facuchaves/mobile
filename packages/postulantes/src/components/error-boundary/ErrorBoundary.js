import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export class ErrorBoundary extends PureComponent {
  state = { hasError: false }

  static getDerivedStateFromError() {
    return { hasError: true }
  }

  componentDidCatch(error, errorInfo) {
    console.error(`Error desde did catch de ErrorBoundary. Error : ${error} , info : ${errorInfo.componentStack}`)
  }

  render() {
    const { renderOnError, children } = this.props
    const { hasError } = this.state

    if (hasError) {
      return renderOnError()
    }

    return children
  }
}

const DefaultErrorScreen = () => (
  <div>
    <h2>Algo no sali√≥ bien</h2>
  </div>
)

ErrorBoundary.propTypes = {
  renderOnError: PropTypes.func,
  children: PropTypes.node.isRequired,
}

ErrorBoundary.defaultProps = {
  renderOnError: DefaultErrorScreen,
}
