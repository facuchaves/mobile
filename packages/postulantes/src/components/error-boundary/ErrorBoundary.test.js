import React from 'react'
import renderer from 'react-test-renderer'

import { ErrorBoundary } from './ErrorBoundary'

const FailingComponent = () => {
  throw new Error('Fail')
}

describe('ErrorBoundary snapshots', () => {
  it('with no error', () => {
    const tree = renderer.create(<ErrorBoundary>Hola</ErrorBoundary>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('with error', () => {
    // eslint-disable-next-line no-undef
    spyOn(console, 'error')

    const tree = renderer
      .create(
        <ErrorBoundary withBorder>
          <FailingComponent />
        </ErrorBoundary>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
