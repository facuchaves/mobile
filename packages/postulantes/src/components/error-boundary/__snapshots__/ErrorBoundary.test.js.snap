// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary snapshots with error 1`] = `
<div>
  <h2>
    Algo no sali√≥ bien
  </h2>
</div>
`;

exports[`ErrorBoundary snapshots with no error 1`] = `"Hola"`;
